web/src/nav/StatusMessagesNavItem.tsx

```diff
@@ -88,9 +108,31 @@ export class StatusMessagesNavItem extends React.PureComponent<Props, State> {
         this.subscriptions.unsubscribe()
     }
 
-    private renderMessage(message: GQL.IStatusMessage): JSX.Element | null {
-        switch (message.type) {
-            case GQL.StatusMessageType.CLONING:
+    private renderSyncErrorMessage(message: GQL.ISyncErrorStatusMessage): JSX.Element | null {
```

Comment:
This is terser, and avoids declaring `displayName` and `extSvcID`.

```suggestion
    private renderSyncErrorMessage({ message, externalServiceDisplayName, externalServiceId}: GQL.ISyncErrorStatusMessage): JSX.Element | null {
```

Overall though, I feel like this doesn't need to be a separate function and could just be inlined.